data/lib/api/match/match_model.dart

```diff
@@ -346,128 +348,67 @@ enum MatchGroup {
   const MatchGroup(this.value);
 }
 
-@freezed
-class TeamMatchStatus with _$TeamMatchStatus {
-  const factory TeamMatchStatus({
-    @Default(0) int win,
-    @Default(0) int tie,
-    @Default(0) int lost,
-  }) = _TeamMatchStatus;
-}
+extension SingleMatchTeamStatExtension on MatchModel {
+  TeamStat updateTeamStat(String teamId, TeamStat currentStat) {
+    final team = _getTeam(teamId);
+    final opponentTeam = _getOpponentTeam(teamId);
 
-@freezed
-class TeamStat with _$TeamStat {
-  const factory TeamStat({
-    @Default(0) int played,
-    @Default(TeamMatchStatus()) TeamMatchStatus status,
-    @Default(0) int runs,
-    @Default(0) int wickets,
-    @Default(0.0) double batting_average,
-    @Default(0.0) double bowling_average,
-    @Default(0) int highest_runs,
-    @Default(0) int lowest_runs,
-    @Default(0.0) double run_rate,
-  }) = _TeamStat;
-}
-
-extension TeamStatExtension on List<MatchModel> {
-  TeamStat teamStat(String teamId) {
-    if (isEmpty) return const TeamStat();
+    final matchRuns = team.run;
+    final matchWickets = team.wicket;
+    final battingAverage =
+        opponentTeam.wicket > 0 ? matchRuns / opponentTeam.wicket : 0.0;
+    final bowlingAverage =
+        matchWickets > 0 ? opponentTeam.run / matchWickets : 0.0;
 
-    var runs = 0;
-    var wickets = 0;
-    var battingAverageTotal = 0.0;
-    var bowlingAverageTotal = 0.0;
-    var highestRuns = 0;
-    var lowestRuns = double.maxFinite.toInt();
+    final result = _matchResult(teamId, team.run, opponentTeam.run);
 
-    for (final match in this) {
-      final team = _getTeam(match, teamId);
-      final opponentTeam = _getOpponentTeam(match, teamId);
+    final totalOvers = currentStat.run_rate > 0
+        ? _computeOvers(currentStat.runs, currentStat.run_rate)
+        : 0.0;
 
-      runs += team.run;
-      wickets += team.wicket;
-
-      battingAverageTotal +=
-          opponentTeam.wicket > 0 ? team.run / opponentTeam.wicket : 0;
-      if (team.wicket > 0) {
-        bowlingAverageTotal += opponentTeam.run / team.wicket;
-      }
-
-      if (team.run > highestRuns) highestRuns = team.run;
-      if (team.run < lowestRuns) lowestRuns = team.run;
-    }
-
-    final bowlingAverage = length > 0.0 ? bowlingAverageTotal / length : 0.0;
+    final matchOvers = team.over;
+    final updatedOvers = totalOvers + matchOvers;
```

Comment:
is this addition of over valid? 2.6 + 1.2 = 3.8 as per this but as per over it should be 4.2, isn't it?? if so, use functions that we've defined to add overs in **data/lib/extensions/double_extension.dart**